fastfield_codecs/src/line.rs

```diff
@@ -69,11 +69,17 @@ impl Line {
 
     // Same as train, but the intercept is only estimated from provided sample positions
     pub fn estimate(ys: &dyn Column, sample_positions: &[u64]) -> Self {
-        Self::train_from(ys, sample_positions.iter().cloned())
+        Self::train_from(
+            ys,
+            sample_positions
+                .iter()
+                .cloned()
+                .map(|pos| (pos, ys.get_val(pos))),
```

Comment:
It would be much more efficient to remove the `get_val` call and use the iterator here.

Reply:
(that requires removing `Line::estimate` and call `Line::train_from` directly in the codec.)

Reply:
This are only 20 samples, but span the whole Column, so I'm not so sure this is more efficient. It depends on the iterator implementation if it can skip efficiently.

Reply:
The default impl calls `next()` https://doc.rust-lang.org/src/core/iter/traits/iterator.rs.html#284-289, that's probably much slower for large indexes.